lm-agent-snap/hooks/bin/configure

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+"""Snapcraft `configure` hook for the License Manager Agent snap."""
+
+import os
+import subprocess
+import sys
+from pathlib import Path
+from typing import Union
+
+SNAP_COMMON_PATH = "/var/snap/license-manager-agent/common"
+SNAP_INSTANCE_NAME = os.environ["SNAP_INSTANCE_NAME"]
+DOTENV_PREFIX = "LM_AGENT_"
+DOTENV_FILE_LOCATION = Path(f"{SNAP_COMMON_PATH}/.env")
+AGENT_VARIABLES_MAP: dict[str, Union[str, int]] = {
+    "BACKEND_BASE_URL": "https://apis.vantagehpc.io",
+    "OIDC_DOMAIN": "auth.vantagehpc.io/realms/vantage",
+    "OIDC_CLIENT_ID": "",
+    "OIDC_CLIENT_SECRET": "",
+    "SACCTMGR_PATH": "/usr/bin/sacctmgr",
+    "SCONTROL_PATH": "/usr/bin/scontrol",
+    "SQUEUE_PATH": "/usr/bin/squeue",
+    "LMUTIL_PATH": "/usr/local/bin/lmutil",
+    "RLMUTIL_PATH": "/usr/local/bin/rlmutil",
+    "LSDYNA_PATH": "/usr/local/bin/lsdyna",
+    "LMXENDUTIL_PATH": "/usr/local/bin/lmxendutil",
+    "OLIXTOOL_PATH": "/usr/local/bin/olixtool",
+    "RESERVATION_IDENTIFIER": "license-manager-reservation",
+    "LM_USER": "license-manager",
+    "STAT_INTERVAL": 60,
+    "TOOL_TIMEOUT": 6,
+    "ENCODING": "utf-8",
+}
+
+
+def daemon_starter():
+    """Start the daemon."""
+    try:
+        run_bash(f"snapctl start {SNAP_INSTANCE_NAME}.daemon")
+    except Exception:
+        sys.exit(1)
+
+
+def daemon_stopper():
+    """Stop the daemon."""
+    try:
+        run_bash(f"snapctl stop {SNAP_INSTANCE_NAME}.daemon")
+    except Exception:
+        sys.exit(1)
+
+
+def snapctl_get(snap_config_value: str) -> Union[str, None]:
+    """Get snap config from snapctl.
+
+    Return python None if snapctl returns the empty string.
+    """
+    snapctl_out: Union[str, None]
```

Comment:
Why use the old style type definintion here instead of `str | None`?

Reply:
I don't know why but the type checker failed sometimes for the vantage agent. Here I just copied tbh.

Reply:
I also remember having errors after the snap got installed due to python versions not accepting this. For example, installing the snap on ubuntu 20.04 would fail

Reply:
I guess this must be due to the snap using the system python. For 20.04, it's Python 3.8. So, this would explain why there are problems. Thanks for the explanation!